Editor/Scripts/interactivetutorials_dialog.py

```diff
@@ -229,15 +254,51 @@ def load_next_step(self):
 
     def load_previous_step(self):
         if self.current_step:
+            self.current_step_index -= 1
             prev_step = self.current_step.prev_step
             if prev_step:
+                self.current_step_index -= 1
                 self.load_step(prev_step)
 
+
     def on_start_button_clicked(self):
         tutorial_index = self.tutorial_list.currentIndex().row()
 
         self.load_tutorial(tutorial_index)
 
+    def find_child_in_hierarchy_with_index(self, obj_parent, obj_index, pattern):
+        """
+        Searches for a hierarchy of children descending from a parent.
+        obj_parent: The Qt object to search within. If None, this will search all top level windows.
+        obj_index: The index of the Qt object to return. Often, an obj_parent has several children of the same type. 
+        patterns: the type of the unnamed Qt object to highlight.
+        The patterns are tested in order.
+
+        For example, to look for the second QComboBox in a hierarchy like the following:
+
+        QWidget (window)
+            -QTabWidget
+                -QWidget named "m_exampleTab"
+                    -QComboBox
+                    -QComboBox
+                    -QComboBox
+
+        Invoke:
+        find_child_in_hierarchy_with_index("m_exampleTab", 1, QtWidgets.QComboBox)
+
+        
+        """
+
+        current_objects = pyside_utils._get_parents_to_search(obj_parent)
+
+        candidates = []
+        for parent_candidate in current_objects:
+            candidates += pyside_utils.find_children_by_pattern(parent_candidate, pattern=pattern, recursive=False)
+        if len(candidates) == 0:
+            return None
+        current_objects = candidates
+
+        return current_objects[obj_index]
```

Comment:
This is awesome! I think this would actually be beneficial to put in `pyside_utils` itself so it could be used in other things like automated testing. The `pyside_utils` does have a `find_child_by_hierarchy` which operates similarly, but always returns the first object in the list, so maybe you could modify that API to take an optional index?

Reply:
Yes, I didn't want to modify pyside because I didn't want to break any tests :) I think this is very basic and could be improved.

Reply:
Not sure but I think the pyside_utils version is only used in one test, so maybe I shouldn't be so concerned about breakage,

Reply:
Looks like it's only used in 3 tests:

```
AutomatedTesting\Gem\PythonTests\editor\EditorScripts\AssetBrowser_TreeNavigation.py 
AutomatedTesting\Gem\PythonTests\editor\EditorScripts\EntityOutliner_EntityOrdering.py
AutomatedTesting\Gem\PythonTests\scripting\NodePalette_HappyPath_CanSelectNode.py
```
An AR run should catch any breakage, so you should be safe. Or we could add a follow-up task to move this functionality into `pyside_utils` after we move its location. I'm good either way!